File: lib/db/models/Registration.ts
import mongoose, { Schema, Document } from "mongoose"

export interface IRegistration extends Document {
  applicationType: "membership" | "leadership"
  name: string
  gender: string
  registrationNo: string
  course: string
  age: number
  instagramId?: string
  mobileNo: string
  whatsappNo: string
  emailId: string
  religion: "catholic" | "christian" | "other"
  address: string
  skills?: string[]
  otherSkills?: string
  eventExperience?: string
  justSocietyDefinition?: string
  communicationExample?: string
  aicufVision?: string
  leadershipPosition?: string
  declaration: boolean
  additionalMessage?: string
  registrationId: string
  status: "pending" | "approved" | "rejected"
  createdAt: Date
  updatedAt: Date
}

const RegistrationSchema = new Schema<IRegistration>(
  {
    applicationType: {
      type: String,
      required: true,
      enum: ["membership", "leadership"],
    },
    name: {
      type: String,
      required: true,
    },
    gender: {
      type: String,
      required: true,
    },
    registrationNo: {
      type: String,
      required: true,
    },
    course: {
      type: String,
      required: true,
    },
    age: {
      type: Number,
      required: true,
    },
    instagramId: String,
    mobileNo: {
      type: String,
      required: true,
    },
    whatsappNo: {
      type: String,
      required: true,
    },
    emailId: {
      type: String,
      required: true,
      lowercase: true,
    },
    religion: {
      type: String,
      required: true,
      enum: ["catholic", "christian", "other"],
    },
    address: {
      type: String,
      required: true,
    },
    skills: [String],
    otherSkills: String,
    eventExperience: String,
    justSocietyDefinition: String,
    communicationExample: String,
    aicufVision: String,
    leadershipPosition: String,
    declaration: {
      type: Boolean,
      required: true,
    },
    additionalMessage: String,
    registrationId: {
      type: String,
      required: true,
      unique: true,
    },
    status: {
      type: String,
      enum: ["pending", "approved", "rejected"],
      default: "pending",
    },
  },
  {
    timestamps: true,
  }
)

RegistrationSchema.index({ emailId: 1 })
RegistrationSchema.index({ registrationId: 1 })
RegistrationSchema.index({ createdAt: -1 })

export default mongoose.models.Registration || mongoose.model<IRegistration>("Registration", RegistrationSchema)

File: backend/app/schemas/registration.py
from typing import Optional, List, Any
from datetime import datetime
from pydantic import BaseModel, EmailStr

class RegistrationBase(BaseModel):
    application_type: str
    name: str
    gender: str
    registration_no: str
    course: str
    age: int
    instagram_id: Optional[str] = None
    mobile_no: str
    whatsapp_no: str
    email_id: EmailStr
    religion: str
    address: str
    
    skills: Optional[List[str]] = []
    other_skills: Optional[str] = None
    event_experience: Optional[str] = None
    
    just_society_definition: Optional[str] = None
    communication_example: Optional[str] = None
    yesj_vision: Optional[str] = None
    leadership_position: Optional[str] = None
    
    declaration: bool = False
    additional_message: Optional[str] = None

class RegistrationCreate(RegistrationBase):
    password: str

class RegistrationUpdate(BaseModel):
    status: str

class Registration(RegistrationBase):
    id: int
    status: str
    created_at: datetime
    
    # We don't return password
    
    class Config:
        orm_mode = True
